DataDog/libdatadog
# Intern stacktraces to reduce memory usage

# What does this PR do?

Interns the stack trace used by the profiler.  This still uses samples: FxIndexMap<Sample, Vec<i64>> for storage, but Samples now store a StackTraceId instead of Vec<PProfId> (where the ids are for locations). This makes it so when two samples have the same stack trace but different labels, they don't end up duplicating the vector of location ids, and instead share the same StackTraceId.



# Motivation

Currently, multiple samples with the same stack trace end up repeatedly paying the cost of storing that stack trace. Instead, 

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

I validated the memory savings using the modifier c-ffi test. 

We should run this on some client libraries to ensure that it doesn't break before merging.
